File: C++/프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

void dfs(int cur, vector<vector<int>> computers, vector<bool>& visited);

int solution(int n, vector<vector<int>> computers)
{
	int answer = 0;
	vector<bool> visited(n, false);

	for (int i = 0; i < n; i++)
	{
		if (!visited[i])
		{
			answer++;
			dfs(i, computers, visited);
		}
	}

	return answer;
}

void dfs(int cur, vector<vector<int>> computers, vector<bool>& visited)
{
	visited[cur] = true;

	for (int nxt = 0; nxt < computers.size(); nxt++)
	{
		if (!visited[nxt] && computers[cur][nxt] == 1)
		{
			dfs(nxt, computers, visited);
		}
	}
}

File: C#/프로그래머스/3/43162. 네트워크/네트워크.cs
using System;

public class Solution {
    public void dfs(int cur, int[,] computers, bool[] visited) {
        visited[cur] = true;
        
        for(int i = 0; i < computers.GetLength(0); i++) {
            if(computers[cur, i] == 1 && !visited[i]) {
                dfs(i, computers, visited);
            }
        }
    }
    
    public int solution(int n, int[,] computers) {
        int answer = 0;
        
        bool[] visited = new bool[n];
        
        for(int i = 0; i < n; i++) {
            if(!visited[i]) {
                dfs(i, computers, visited);
                answer++;
            }
        }
        
        return answer;
    }
}